### PR TITLE
refactor: clean code

### PR DESCRIPTION
**Pull request by Assstra**
# Pull Request: Refactor/clean code

## Description:
The goal of this PR is to introduce the implementation of the TryFrom trait for the RustAgent struct. Moreover, the point is also to clear the code, cleaning unwrap and other bad practices.

## Changes:
Todo : 
- [x] Changed From to TryFrom (Rust agent)
- [ ] Handle some unwrap()
- [ ] ...

## Testing:
As before

### CODE REVIEW

File: src/agent/src/workload/runner.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub struct Runner {
 impl Runner {
     pub fn new(config: Config) -> Self {
         let agent: Box<dyn Agent + Sync + Send> = match config.language {
-            Language::Rust => Box::new(rust::RustAgent::from(config.clone())),
+            Language::Rust => Box::new(rust::RustAgent::try_from(config.clone()).expect("Failed to convert Config to RustAgent")),
```
Comment (mfernd) [inline src/agent/src/workload/runner.rs:22]: `expect()` panics so it's like an `unwrap()`

Comment (Assstra) [reply src/agent/src/workload/runner.rs:22]: That's true. I wanted to do something like this :
```rust
let agent = match rust::RustAgent::try_from(config.clone()) {
    Ok(agent) => Box::new(agent),
    Err(err) => {
       // I don't know how to handle the error here as propagating is not an option
    }
};
```

Comment (mfernd) [reply src/agent/src/workload/runner.rs:22]: so we need to return a result for `Runner::new()`

and you don't need the `match` if the `RustAgent` `try_from` already returns an `AgentError` (to change in the Agent also)

**Comment by Assstra** [comment]
As we discussed, sounds like it would be too much work to refactor everything up to tonight :/

**Pull request closed without merging**